gcp/compute_engine/resource_policies/disk_snapshot_schedule/README.md

```diff
@@ -0,0 +1,83 @@
+# Disk snapshot schedule
+This module creates a disk snapshot schedule, and can optionally attach the schedule to a set of disks.
+
+Required variables:
+* `project` - GCP project
+* `name` - Name of the snapshot schedule
+* `region` - Snapshot schedule region. If attaching snapshot schedule to regional compute disks, 
+             the disks must be in this region.
+
+Other variables:
+* The `retention_policy` variable is a map defining how long a snapshot should be kept and has:
+  * `max_retention_days` - Number of days to keep the variable
+  * `on_source_disk_delete` - Specifies what to do to a scheduled snapshot if the disk it is attached to is deleted. 
+     Possible values are:
+    * `KEEP_AUTO_SNAPSHOTS` (default) 
+    * `APPLY_RETENTION_POLICY`
+* The `snapshot_properties` variabled is a map defining configuration for the snapshot:
+    * `labels` - snapshot labels
+    * `storage_locations` - Cloud storage bucket location to autostore snapshot (e.g. `["europe-west2"]`)
+    * `guest_flush` - Whether to flush all pending writes from memory to disk before the snapshot is capture
+* The `shapshot_schdeule` variable is a map defining when snapshots should be taken, and has the following values:  
+  * `frequency` - One of `"hourly"` `"daily"` or `"weekly"` (Default `"daily"`)
+  * `interval` - If using `hourly` or `daily` snapshots, defines the interval between snapshots (Default `1`)
+  * `weekly_snapshot_schedule` - If doing weekly, include a list of mapptings with the `day` and `start_time` of the snapshot
+
+
+
+Example `snapshot_schedule` configurations:
+```terraform
+snapshot_schedule = {
+  frequency = "daily"
+  interval = number
+  start_time = time
+}
+snapshot_schedule = {
+  frequency = "weekly"
+  weekly_snapshot_schedule = [
+    {
+      day = "TUESDAY"
+      start_time = "23:00"
+    },
+    {
+      day = "FRIDAY"
+      start_time = "03:00"
+    }
+  ]
+}
+```
+
+### Attaching snapshot to disks
+If there are existing disks that the created snapshot schedule should be applied to, 
+they can be specified and managed by the `google_compute_disk_resource_policy_attachment` resource.
+To specify disks the schedule should be attached to list define:
+* `disks` - A map with the disk name as the key, and it's zone as the value (**NOTE** disks must be in the same region). 
+  E.g. 
+  ```terraform
+  disks = {
+    disk1 = "europe-west2-a"
+    disk2 = "europe-west2-b"
+  }
+  ```
+* `regional disk` - A list of regional disks within the same region as the snapshot schedule
+
+### Destroying resource
+_**NOTE**: A snapshot schedule cannot be deleted if it is attached to a disk._  
```

Comment:
Ok that makes sense. 
Issue is, this was separated from the secure-swarm module because disks weren't being specified, therefore when doing `terraform destroy`, there would be an error because the `google_compute_resource_policy` was attached to the disk still, leaving the `google_compute_resource_policy` as the remaining resource in the terraform state. So this would be an expected behaviour? 
I guess for our use of secure-swarm module, `terraform destroy` would not be done often, so it wouldn't effect much, but if we were to want to update the snapshot schedule, we would have to manually detach, update the policy, and then manually attach the policy.
Removing the attachment resource from this module would mean a rethink about how we would handle the secure-swarm use case, as having 3 separate terraform states (secure-swarm, resource policy, and resource policy attachment) for a project, doesn't seem very practical

Reply:
Why would `google_compute_resource_policy` be in the `secure_swarm` state in the first place? At most I would expect only the attachment resource; and in our case, with the disks being only managed by the platform, not terraform, probably not even that.

Reply:
Removed attachment resource, and added a bit of documentation to secure-swarm readme